js/enter_game.js

```diff
@@ -1,81 +1,134 @@
+  } else return
+
+  document.getElementById(player).disabled = !checkbox.checked
+  // enable/disable tokenize2 instance of player explicitly
+  const tok2p2 = tok2.find(tok2p => {
+    return tok2p.element[0].id === player
+  })
+  if (checkbox.checked) {
+    tok2p2.enable()
   } else {
-    hideElment('playerA2', true)
-    isAhidden = true
+    tok2p2.clear()
+    tok2p2.disable()
   }
 }
 
-function teamBHas2ndPlayer () { // eslint-disable-line no-unused-vars
-  // textinput B
-  if (isBhidden) {
-    hideElment('playerB2', false)
-    isBhidden = false
-  } else {
-    hideElment('playerB2', true)
-    isBhidden = true
+async function differentPlayersEntered (data) { // eslint-disable-line no-unused-vars
+  // create list of entered players
+  let playerIDList = [data.playerA1, data.playerA2, data.playerB1, data.playerB2]
+  // remove null from list of entered players -> list of actual players
+  playerIDList = playerIDList.filter(function (value, index, array) {
+    return value !== null
+  })
+  // create set of actual players -> remove duplicate entries (little reminiscence of The Exmatriculator)
+  const playerIDSet = new Set(playerIDList)
+  // entered players are all different, if list of actual players contains as much elements as set of actual players
+  return playerIDList.length === playerIDSet.size
+}
+
+async function validateUserInput (data) { // eslint-disable-line no-unused-vars
+  // check if enabled players are entered
+  if ((!document.getElementById('playerA2').disabled && !data.playerA2) || (!document.getElementById('playerB2').disabled && !data.playerB2)) {
+    console.log('Enter missing players or disable them!')
+    return false
+  }
+
+  // check if entered players are all different
+  if (!await differentPlayersEntered(data)) {
+    console.log('Choose different players!')
+    return false
+  }
+
+  // check if match result (goals per team) is entered
+  if (data.goalsA === null || data.goalsB === null) {
+    console.log('Enter a complete match result!')
+    return false
+  }
+
+  return true
+}
+
+async function Submit () { // eslint-disable-line no-unused-vars
+  // collect entered data
+  const data = {
+    action: 'enterGame',
+    playerA1: document.getElementById('playerA1').value,
+    playerB1: document.getElementById('playerB1').value,
+    playerA2: !document.getElementById('playerA2').disabled ? document.getElementById('playerA2').value : null,
+    playerB2: !document.getElementById('playerB2').disabled ? document.getElementById('playerB2').value : null,
+    goalsA: document.getElementById('goalsAInput').value ? document.getElementById('goalsAInput').value : null,
+    goalsB: document.getElementById('goalsBInput').value ? document.getElementById('goalsBInput').value : null
+  }
+
+  // validate entered data
+  if (!await validateUserInput(data)) {
+    return
+  }
+
+  const res = await apiPost(data)
+  if (res.success) {
+    console.log('New match was successfully added!')
```

Comment:
This should either redirect somewhere (`/matches`?) or clear the inputs. Otherwise a match can be entered multiple times by accident

Reply:
Something like this: https://github.com/DavidRisch/kicker/blob/c2a6eb3c203f0e911507b95dde77b717e321d1bc/js/edit_account.js#L55